internal/auth/repository_auth_method.go

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package auth
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/db/timestamp"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/kms"
+	"github.com/hashicorp/boundary/internal/pagination"
+	"github.com/hashicorp/boundary/internal/util"
+	"github.com/hashicorp/go-kms-wrapping/v2/extras/structwrapping"
+)
+
+// AuthMethodRepository coordinates calls across different subtype services
+// to gather information about all auth methods.
+type AuthMethodRepository struct {
+	reader db.Reader
+	writer db.Writer
+	kms    *kms.Kms
+}
+
+// NewAuthMethodRepository returns a new auth method repository.
+func NewAuthMethodRepository(ctx context.Context, reader db.Reader, writer db.Writer, kms *kms.Kms) (*AuthMethodRepository, error) {
+	const op = "auth.NewAuthMethodRepository"
+	switch {
+	case util.IsNil(reader):
+		return nil, errors.New(ctx, errors.InvalidParameter, op, "missing DB reader")
+	case util.IsNil(writer):
+		return nil, errors.New(ctx, errors.InvalidParameter, op, "missing DB writer")
+	}
+	return &AuthMethodRepository{
+		reader: reader,
+		writer: writer,
+		kms:    kms,
+	}, nil
+}
+
+// List lists auth methods across all subtypes.
+func (amr *AuthMethodRepository) List(ctx context.Context, scopeIds []string, afterItem pagination.Item, limit int) ([]AuthMethod, time.Time, error) {
```

Comment:
Does this need to be exported?

Reply:
Leaving them unexported brings in an import cycle in the subtype packages